core/libs/meta/migration/driver.php

```
@@ -62,22 +64,35 @@ public function migrate()
             foreach ($iterator as $fileinfo) {
                 if ($fileinfo->isFile()) {
                     if (preg_match('/\.php$/', $fileinfo->getFilename())) {
-                        $model = str_replace('.php', '', basename($fileinfo->getFilename()));
-                        $result[$model] = 'Failed to migrate '.$model;
-                        $this->di->get('metaMigration')->migrateModel($model);
-                        $result[$model] = $model." migrated successfully";
+                        $models[] = str_replace('.php', '', basename($fileinfo->getFilename()));
                     }
                 }
             }
-       } catch (PhalconException $e) {
-           print($e->getMessage());
-       } catch (Exception $e) {
-           if ($extensionLoaded) {
-               print Color::error($e->getMessage()) . PHP_EOL;
-           } else {
-               print 'ERROR: ' . $e->getMessage() . PHP_EOL;
-           }
-       }
-       return $result;
+
+            // before migrating table or views, let's migrate functions. Because table or views use
+            // functions. So migrating functions is required first.
+            foreach ($models as $model) {
+                $this->di->get('metaMigration')->migrateFunctions($model);
+            }
+
+            //migrate models
+            foreach ($models as $model) {
+                $result[$model] = "Failed to migrate $model";
+                $this->di->get('metaMigration')->migrateModel($model);
+                $result[$model] = "$model migrated successfully";
+            }
+        }
+        catch (PhalconException $e) {
+            print($e->getMessage());
+        }
+        catch (Exception $e) {
+            if ($extensionLoaded) {
+                print Color::error($e->getMessage()) . PHP_EOL;
```

Lets throw the exception for 500 and let the index handle the output.